### ISSUE TITLE
Problem in predictions

### ISSUE THREAD

**Issue by jessiffmm** [pierluigiferrari/ssd_keras#210]
Hi!!!

I trained the ssd300 model with my datasets. But I have a problem. When I do the predictions I only get 2 detections. I mean, for example I have a image with a lot objects, but my model only detect two object. I don't undersatand why It happens.

Regards

**Comment by amichayfeldman** [comment]
You have to change hyperparameters and continue with the training or investigate if data processing will help you.

However, i faced with errors when try to train ssd300 and ssd512. Did you chage some "internal" functions? my problems occurs with gradients computations  (i open issue about this).

**Comment by jessiffmm** [comment]
I tried the example that indicates of ssd300. With ssd7 I get a lot of detections but with ssd300 only two detections. I didn't change anything.

**Comment by pierluigiferrari** [comment]
In `decode_detections()`, have you set `top_k=2` by any chance? That's the only reason I can think of why you would never get more than 2 predictions.

**Comment by jessiffmm** [comment]
No, I have 200. I have seen that it only detects an object of the same class. I changed decode_detections() by decode_detections_fast and I haven't this problem.
